ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -141,13 +141,60 @@ def get_encoder_for_obs(
             return (visual_encoder_class(shape[0], shape[1], shape[2], h_size), h_size)
         raise UnityTrainerException(f"Unsupported shape of {shape} for observation")
 
+    @staticmethod
+    def can_encode_visual(sensor_spec: ObservationSpec) -> bool:
+        """
+        Returns True if it is possible to create a visual embedding for the sensor
+        """
+        if len(sensor_spec.shape) != 3:
```

Comment:
if all of the dim properties were unspecified, this would return true? That feels wrong.

Reply:
To be compatible with previous environments, I need the behavior to be convolutions when the properties are unspecified. so I need each dimension to be either translational_equiv or unspecified.